flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/MemoryArchivist.scala

```diff
@@ -183,6 +193,33 @@ class MemoryArchivist(private val max_entries: Int)
     }
   }
 
+  private def archiveJsonFiles(graph: ArchivedExecutionGraph) {
+    future {
+      val rootPath = new Path(flinkConfiguration.getString(
```

Comment:
Since this is executed async, we should add a safety try-catch around it that logs any exceptions